### PR TITLE
[WIP] feat(core): use npm commands

### PR DESCRIPTION
**Pull request by simison**
Replace variety of gulp/grunt/sh/npm commands with npm commands.

Todo:
- [x] Add commands
- [ ] Test
- [ ] Test with Docker
- [ ] Update readme

See #1258 for discussion.

### CODE REVIEW

**Comment by simison** [comment]
Notes:
```
"test:watch:onlyChanged": "gulp test:server:watch --onlyChanged"
```
npm commands do support parameters, but it's kinda cumbersome (`npm command -- --param`)

```
"prestart": "if [ ! -d node_modules ]; then npm install --quiet; fi"
```
So no need to run `npm install`, just clone and `npm start` will do. Thoughts?

```
"postinstall": "bower install --allow-root --config.interactive=false && bower prune --allow-root --config.interactive=false"
```
`--allow-root` helps with docker and CI environments. Note that it doesn't turn sudo on, it merely allows it. Normally Bower would complain about it. This sorta just moves responsibility to developer, assuming they should know what they're doing with user rights. Thoughts? These config vars could of course be moved to [`.bowerrc`](https://github.com/meanjs/mean/blob/master/.bowerrc) but I think here they're more clear.

File: package.json [hunk 1]
```diff
@@ -15,9 +15,20 @@
...
+    "start:prod": "gulp prod",
+    "start:debug": "gulp debug",
+    "lint": "gulp lint",
+    "test": "gulp test",
+    "test:server": "gulp test:server",
+    "test:client": "gulp test:client",
+    "test:e2e": "gulp test:e2e",
+    "test:watch": "gulp test:server:watch",
+    "test:watch:onlyChanged": "gulp test:server:watch --onlyChanged",
+    "prestart": "if [ ! -d node_modules ]; then npm install --quiet; fi",
```
Comment (ilanbiala) [inline package.json]: Too black box magic for me, I don't think we should do this.

Comment (codydaig) [inline package.json]: @ilanbiala All the commands in general or just the prestart command?

Comment (ilanbiala) [inline package.json]: prestart

Comment (codydaig) [inline package.json]: I'm not opposed to it being in there, but don't see any reasons to have it.

Comment (ilanbiala) [inline package.json]: I feel like people are gonna ask questions about it and it's gonna confuse beginners more than help.

Comment (codydaig) [inline package.json]: @simison Go ahead and remove this command. There's really no benefit to using this vs. npm install. Plus, this doesn't cover the edge case where a module was added or failed to install and npm install needs to be ran again even though node_modules directory exists.

Comment (simison) [inline package.json]: 👍  done!

**Comment by ilanbiala** [comment]
Also, can we move the bower flags into a bowerrc?

**Comment by simison** [comment]
>Also, can we move the bower flags into a bowerrc?

Done.

**Comment by ilanbiala** [comment]
Looks a lot cleaner to me :+1:

**Comment by trendzetter** [comment]
:+1:
also wasn't there the new gulp-wiredep or something?

**Comment by mleanos** [comment]
We could simplify the `scripts` section in the package.json by only including the tasks necessary for install, testing, and deployment; we don't need all the tasks that are solely used for development. IMO, the only testing task we need is `test`.

If we add a script entry of `"gulp": "gulp"` (uses the locally installed version of Gulp), we could run any other Gulp task defined in `gulpfile.js` with `npm run gulp test:server:watch`. This could be very useful for our users, since they may have a list of custom tasks that they would want to continue to use. If a user prefers to use Grunt, or a mix of Gulp & Grunt, then they could add an entry for Grunt to the scripts section.

**Comment by mleanos** [comment]
@simison Any thoughts on my previous comment?

**Comment by codydaig** [comment]
Bump. Any progress with this PR?

**Comment by lirantal** [comment]
@codydaig  I definitely want to see this through! I'll get back to it if @simison won't be continuing this PR.

**Comment by mleanos** [comment]
@codydaig @lirantal Any thoughts on [my comment](https://github.com/meanjs/mean/pull/1296#issuecomment-215896613)?

**Comment by codydaig** [comment]
@mleanos IMO, I would like to see the most common tasks put in the scripts section of package.json (ie. watch). Tasks that are only typically ran ocassionally like seeding of the database, can be left to the gulp/grunt file.

**Comment by lirantal** [comment]
@codydaig , guys this LGTM and I want to merge it in with a follow-up PR for updating the README and docs. I don't think we need to "simplify" it. It's not complex if it is documented and we get people used to use the `npm` command.

@mleanos I get what you mean with making the `gulp` task more portable. I don't think it conflicts, we can also add that kind of command to the script and win both worlds.

WDYS?

**Comment by mleanos** [comment]
@lirantal That sounds good to me. Perhaps we could add the entry for Gulp, with the follow up PR with an explanation of this in the docs.

In that case, this PR LGTM.

**Comment by simison** [comment]
Excellent, thanks @codydaig and @mleanos for taking care of this! I'm super busy myself right now.

@mleanos FYI, this changed since this PR: https://github.com/meanjs/mean/commit/eaead7a30cb2a60ad25250940e79b42619261e23

**Comment by mleanos** [comment]
@simison Do you see the commit that you referenced as a conflict, or an issue?

From my perspective neither this commit, or the changes to the `test:server:watch` Gulp task have an impact over the other. The task name hasn't changed, but the internal task used within the Gulp config has; this should be OK though.

**Pull request merged**